packages/backend/src/models/TagsModel.ts

```diff
@@ -57,4 +58,91 @@ export class TagsModel {
+            const projectYamlTags = await trx(TagsTableName)
+                .whereNotNull('yaml_reference')
+                .andWhere('project_uuid', projectUuid);
+
+            // get all yaml tags that are not in the project but are in the yamlTags array
+            const yamlTagsToCreate = yamlTags.filter(
+                (yamlTag) =>
+                    !projectYamlTags.some(
+                        (projectYamlTag) =>
+                            projectYamlTag.yaml_reference ===
+                            yamlTag.yaml_reference,
+                    ),
+            );
+
+            // get all yaml tags that are in the project but are not in the yamlTags array
+            const yamlTagsToDelete = projectYamlTags.filter(
+                (projectYamlTag) =>
+                    !yamlTags.some(
+                        (yamlTag) =>
+                            yamlTag.yaml_reference ===
+                            projectYamlTag.yaml_reference,
+                    ),
+            );
+
+            // get all updates for the existing project yaml tags
+            const yamlTagUpdates = yamlTags.reduce<Record<string, DbTagUpdate>>(
+                (acc, yamlTag) => {
+                    const projectYamlTag = projectYamlTags.find(
+                        (projectTag) =>
+                            projectTag.yaml_reference ===
+                            yamlTag.yaml_reference,
+                    );
+
+                    if (projectYamlTag) {
+                        acc[projectYamlTag.tag_uuid] = {
+                            color: yamlTag.color,
+                            name: yamlTag.name,
+                        };
+                    }
+
+                    return acc;
+                },
+                {},
+            );
+
+            if (yamlTagsToDelete.length > 0) {
+                // delete all yaml tags that are in the project but are not in the yamlTags array
+                await trx(TagsTableName)
+                    .where('project_uuid', projectUuid)
+                    .whereIn(
+                        'tag_uuid',
+                        yamlTagsToDelete.map((t) => t.tag_uuid),
+                    )
+                    .delete();
+            }
+
+            if (yamlTagsToCreate.length > 0) {
+                // create all yaml tags that are not in the project but are in the yamlTags array
+                await trx(TagsTableName).insert(yamlTagsToCreate);
+            }
+
+            if (Object.keys(yamlTagUpdates).length > 0) {
+                // apply all updates to the existing project yaml tags
+                const updatePromises = Object.entries(yamlTagUpdates).map(
+                    ([tagUuid, tagUpdate]) =>
+                        trx(TagsTableName)
+                            .where('tag_uuid', tagUuid)
+                            .update(tagUpdate),
+                );
+
+                await Promise.all(updatePromises);
```

Comment:
mm, should we do this on a single transaction perhaps ? This can trigger many requests to the database, locking the database..